io/src/read_detector_description.cpp

```diff
@@ -0,0 +1,175 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Library include(s).
+#include "traccc/io/read_detector_description.hpp"
+
+#include "csv/read_surfaces.hpp"
+#include "traccc/io/read_detector.hpp"
+#include "traccc/io/read_digitization_config.hpp"
+#include "traccc/io/utils.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// System include(s).
+#include <sstream>
+#include <stdexcept>
+
+namespace {
+
+void fill_digi_info(traccc::silicon_detector_description::host& dd,
+                    const traccc::module_digitization_config& data) {
+
+    // Set a hard-coded threshold for which cells should be considered for
+    // clusterization on this module / surface.
+    dd.threshold().back() = 0.f;
+
+    // Fill the new element with the digitization configuration for the
+    // surface.
+    const auto& binning_data = data.segmentation.binningData();
+    dd.reference_x().back() = binning_data.at(0).min;
+    dd.reference_y().back() = binning_data.at(1).min;
+    dd.pitch_x().back() = binning_data.at(0).step;
+    dd.pitch_y().back() = binning_data.at(1).step;
+    dd.dimensions().back() = data.dimensions;
+    dd.variance_y().back() = data.variance_y;
```

Comment:
As pointed out, `variance_y` may not be necessary

Reply:
:+1: